### PR TITLE
CLI #10

### PR DESCRIPTION
**Pull request by sankerust**
closes https://github.com/CorpGlory/grafana-plugin-cli/issues/10

Add CLI.
Ask for user input (type in text and multiple choice lists):
image

### CODE REVIEW

File: src/cli.ts [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    message: 'Enter your plugin name:',
+    validate: function (value) {
+      if (value.length) {
+        return true;
+      } else {
+        return 'Please provide at least 1 character as a name.';
+      }
+    }
+  }
+  console.log('-----------------------------');
```
Comment (amper43) [inline src/cli.ts]: Suggest to make function for a separator. It avoids multiple changes if we change separator's text.

File: src/cli.ts [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+  userInput.style = response.style;
+}
+
+export async function runCLI() {
+  console.log('-----------------------------');
+  console.log('Welcome to Grafana-plugin-cli');
+
+  await collectUserInput();
+
+  console.log('For debugging - below is collected user input')
```
Comment (amper43) [inline src/cli.ts]: Maybe it would be better somethnk like `Your configuration`, without `debugging`?

Comment (rozetko) [reply src/cli.ts]: Let's just remove it maybe?:)

File: build/dist/bundle.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
+!function(t){var n={};function e(r){if(n[r])return n[r].exports;var i=n[r]={i:r,l:!1,exports:{}};return t[r].call(i.exports,i,i.exports,e),i.l=!0,i.exports}e.m=t,e.c=n,e.d=function(t,n,r){e.o(t,n)||Object.defineProperty(t,n,{enumerable:!0,get:r})},e.r=function(t){"undefined"!=typeof Symbol&&Symbol.toStringTag&&Object.defineProperty(t,Symbol.toStringTag,{value:"Module"}),Object.defineProperty(t,"__esModule",{value:!0})},e.t=function(t,n){if(1&n&&(t=e(t)),8&n)return t;if(4&n&&"object"==typeof t&&t&&t.__esModule)return t;var r=Object.create(null);if(e.r(r),Object.defineProperty(r,"default",{enumerable:!0,value:t}),2&n&&"string"!=typeof t)for(var i in t)e.d(r,i,function(n){return t[n]}.bind(null,i));return r},e.n=function(t){var n=t&&t.__esModule?function(){return t.default}:function(){return t};return e.d(n,"a",n),n},e.o=function(t,n){return Object.prototype.hasOwnProperty.call(t,n)},e.p="",e(e.s=96)}([function(t,n,e){"use strict";e.d(n,"a",function(){return i});
```
Comment (rozetko) [inline build/dist/bundle.js]: 1. `dist/` should be in a repo root, not inside `build/`
2. Let's add it to `.gitignore` and remove from repo
It makes diffs larger and may be a pain when merging

File: package.json [hunk 4]
```diff
@@ -5,7 +5,8 @@
   "main": "index.ts",
   "scripts": {
     "test": "jest --config jest.config.js",
-    "build": "webpack --mode production --config build/webpack.prod.js"
+    "build": "webpack --mode production --config build/webpack.prod.js",
+    "start": "node ./build/dist/bundle.js"
```
Comment (rozetko) [inline package.json]:
```suggestion
     "start": "node dist/bundle.js"
```

File: src/cli.ts (hunk 1 above)
Comment (sankerust) [reply src/cli.ts]: good idea.

**Pull request merged**